### PR TITLE
ref(replays): update feedback breadcrumb

### PR DESCRIPTION
**Pull request by michellewzhang**
- The new feedback breadcrumb is pink & has a megaphone icon in both the breadcrumbs & timeline.
- The errors count in the top right doesn't include feedback anymore
- Errors tab doesn't include feedback
- Feedback breadcrumb links to the feedback page rather than issues

## New breadcrumb color & icon

SCR-20240320-myzy

SCR-20240320-myze

## No errors in errors tab or the error count in top right:

https://github.com/getsentry/sentry/assets/56095982/46888899-19ae-4f01-b89e-3e1bf7b53708

## Links to the feedback page rather than issues:

https://github.com/getsentry/sentry/assets/56095982/b767c99b-23d0-4097-a92a-934ca4f3af7d

Closes https://github.com/getsentry/team-replay/issues/408

### CODE REVIEW

File: static/app/utils/replays/getFrameDetails.tsx [hunk 1]
```diff
@@ -63,11 +64,17 @@ const MAPPER_FOR_FRAME: Record<string, (frame) => Details> = {
     icon: <IconLocation size="xs" />,
   }),
   issue: (frame: ErrorFrame) => ({
-    color: 'red300',
-    description: frame.message,
+    color: frame.message === 'User Feedback' ? 'pink300' : 'red300',
+    description:
```
Comment (michellewzhang) [inline static/app/utils/replays/getFrameDetails.tsx:68]: we could make this description more interesting, by putting the message preview for example, but that would involve an extra API call to fetch the issue details

Comment (ryan953) [reply static/app/utils/replays/getFrameDetails.tsx:68]: Could we put the message into something like the `error.value` field of the response object?

the response to `/replays-events-meta/` is something like:
```
{
    "data": [
        {
            "issue.id": 5071000502,
            "error.type": "",
            "error.value": "",
            "id": "75deaeb8eb7940ed92d9612c49a1e39a",
            "timestamp": "2024-03-15T19:44:54+00:00",
            "title": "User Feedback",
            "project.name": "pokemart",
            "issue": "POKEMART-8P"
        }
    ],
    "meta": {
        "fields": {
            "issue.id": "integer",
            "error.type": "string",
            "error.value": "string",
            "id": "string",
            "timestamp": "date",
            "title": "string",
            "project.name": "string"
        },
        "units": {
            "issue.id": null,
            "error.type": null,
            "error.value": null,
            "id": null,
            "timestamp": null,
            "title": null,
            "project.name": null
        },
        "isMetricsData": false,
        "isMetricsExtractedData": false,
        "tips": {
            "query": null,
            "columns": null
        },
        "datasetReason": "unchanged"
    }
}
```

cc @JoshFerge

Comment (JoshFerge) [reply static/app/utils/replays/getFrameDetails.tsx:68]: Not easily. we could make a backlog item to address this though, maybe by giving user feedback its own endponit

Comment (ryan953) [reply static/app/utils/replays/getFrameDetails.tsx:68]: ack. kk

File: static/app/utils/replays/hooks/useReplayData.tsx [hunk 2]
```diff
@@ -240,8 +241,12 @@ function useReplayData({
       .concat(extraErrorPages)
       .concat(platformErrorPages)
       .flatMap(page => page.data);
+
```
Comment (michellewzhang) [inline static/app/utils/replays/hooks/useReplayData.tsx:244]: need this to return a count of errors not including UF for our count in the top right

File: static/app/views/replays/details.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ function ReplayDetails({params: {replaySlug}}: Props) {
   // TODO: replayId is known ahead of time and useReplayData is parsing it from the replaySlug
   // once we fix the route params and links we should fix this to accept replayId and stop returning it
   const {
-    errors: replayErrors,
+    errorsNoFeedback,
```
Comment (michellewzhang) [inline static/app/views/replays/details.tsx:57]: updates the counts in the top right

File: static/app/views/replays/details.tsx [hunk 4]
```diff
@@ -67,6 +67,8 @@ function ReplayDetails({params: {replaySlug}}: Props) {
     orgSlug,
   });
 
+  const replayErrors = errors.filter(e => e.title !== 'User Feedback');
```
Comment (michellewzhang) [inline static/app/views/replays/details.tsx:70]: updates the counts in the top right

File: static/app/utils/replays/replayReader.tsx [hunk 5]
```diff
@@ -192,7 +193,9 @@ export default class ReplayReader {
       replayRecord,
       breadcrumbFrames,
       this._sortedRRWebEvents
-    ).sort(sortFrames);
+    )
```
Comment (michellewzhang) [inline static/app/utils/replays/replayReader.tsx:196]: appends `FeedbackFrames` to the list of breadcrumb frames

File: static/app/utils/replays/getFrameDetails.tsx [hunk 6]
```diff
@@ -62,6 +63,13 @@ const MAPPER_FOR_FRAME: Record<string, (frame) => Details> = {
     title: 'Navigation',
     icon: <IconLocation size="xs" />,
   }),
+  feedback: frame => ({
```
Comment (ryan953) [inline static/app/utils/replays/getFrameDetails.tsx:66]:
```suggestion
  feedback: (frame: FeedbackFrame) => ({
```

File: static/app/utils/replays/hydrateErrors.tsx [hunk 7]
```diff
@@ -1,47 +1,79 @@
...
+        message: e.title,
+        offsetMs: Math.abs(time.getTime() - startTimestampMs),
+        timestamp: time,
+        timestampMs: time.getTime(),
+        type: 'error', // For compatibility reasons. See BreadcrumbType
+      });
+      return;
+    } catch (error) {
+      Sentry.captureException(error);
+      return;
```
Comment (ryan953) [inline static/app/utils/replays/hydrateErrors.tsx:74]:
```suggestion
    } catch (error) {
      Sentry.captureException(error);
```

File: static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx [hunk 8]
```diff
@@ -70,6 +70,7 @@ const OPORCATEGORY_TO_TYPE: Record<string, keyof typeof TYPE_TO_LABEL> = {
   'ui.click': 'click',
   'ui.keyDown': 'keydown',
   'ui.input': 'input',
+  feedback: 'feedback',
```
Comment (ryan953) [inline static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx:73]: oh, i'm noticing there is a `'sentry.feedback': 'feedback',` line just above in this same list. what is that?

Comment (michellewzhang) [reply static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx:73]: looks like we added this in here: https://github.com/getsentry/sentry/pull/63883

we originally had this blue breadcrumb showing submitted feedbacks, which i think was coming from the SDK, but i don't see it on new replays anymore. cc @billyvg did these get dropped or do you know why we're not seeing these old feedback breadcrumbs anymore?

Comment (billyvg) [reply static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx:73]: @michellewzhang They should stilll be there (but only the widget feedbacks, not crash reports) https://github.com/getsentry/sentry-javascript/blob/4ba3e405b01c1f37a64bb3c78b96be0a6cfa2571/packages/replay/src/coreHandlers/util/addFeedbackBreadcrumb.ts#L20-L34

Comment (michellewzhang) [reply static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx:73]: weird, i'm not seeing them for widget feedback replays at all

Comment (michellewzhang) [reply static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx:73]: resolved for now with https://github.com/getsentry/sentry/pull/67527: we removed the old breadcrumb

File: static/app/views/replays/details.tsx (hunk 4 above)
Comment (ryan953) [reply static/app/views/replays/details.tsx:70]: those rawErrors leaking out :(

**Review by ryan953 — APPROVED** [review]

**Pull request merged**